### PR TITLE
Ft dev k8s manifests

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
**_WHAT DOES THIS PR DO?_**
- [keep boot disk on instance deletion](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/a4251707acbb11ca534eb2281f221c5a8f837ad3)
- [allow stopping to update](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/a67cd572f2c7b6c301368f0224d36d6aa18a812a)
- [update config servers to 'e2-standard-2'](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/4ba14dcab7cf21af03ad7b91995cb13bf0659bcb)
- [Remove dev sensitive stateful resources](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/35c1f7884d6c0667519815308a665d08da6e1ffb) from terraform
- Add [dev nginx virtual servers and global config](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/fc087057933b3332a12df99c671b2c8ef9d7353f)
- [add nginxinc helm chart installation files](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/80d99974962d59dfc9457a8a4bc106d28d566b50)
- [custom values for nginx installation](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/214eab6e21149239316acf09d7b8678eabfcfba2)
- [switch docs subdomains to wiki](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/bd663a8b432a452925f4de7151f745b2146c223a)
- Add [tls-secret template](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/5c842e8f50fbf1998fb3dcc0387ad4cab08d90af) for nginx

**_HOW DO I TEST OUT THIS PR?_**
- do a static code analysis
- ensure that all the checks pass [especially the one of Terraform](https://app.terraform.io/app/AirQo/workspaces/airqo-250220/runs/run-3EbRRxLgqrovcfVh)
- checkout the docs subdomains and ensure you can securely access them:
  - https://wiki.airqo.net/
  - https://staging-wiki.airqo.net/
  - https://dev-wiki.airqo.net/ (available depending on what time you check)

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- n/a

**_ARE THERE ANY RELATED PRs?_**
- https://github.com/airqo-platform/AirQo-frontend/pull/1316

### CODE REVIEW

File: infra/terraform/gcp/modules/ComputeInstance/airqo-dev-k8s-controller.tf [hunk 1]
```diff
@@ -4,9 +4,11 @@ resource "google_compute_instance" "airqo_dev_k8s_controller" {
   zone    = var.zone["b"]
 
   machine_type = "e2-standard-2"
+  
+  allow_stopping_for_update = true
 
   boot_disk {
-    auto_delete = true
+    auto_delete = false
```
Comment (123MwanjeMike) [inline infra/terraform/gcp/modules/ComputeInstance/airqo-dev-k8s-controller.tf:11]: [keep boot disk on instance deletion](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/a4251707acbb11ca534eb2281f221c5a8f837ad3)

File: infra/terraform/gcp/modules/ComputeInstance/airqo-dev-k8s-controller.tf [hunk 2]
```diff
@@ -4,9 +4,11 @@ resource "google_compute_instance" "airqo_dev_k8s_controller" {
   zone    = var.zone["b"]
 
   machine_type = "e2-standard-2"
+  
+  allow_stopping_for_update = true
```
Comment (123MwanjeMike) [inline infra/terraform/gcp/modules/ComputeInstance/airqo-dev-k8s-controller.tf:8]: [allow stopping to update](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/a67cd572f2c7b6c301368f0224d36d6aa18a812a)

File: infra/terraform/gcp/modules/ComputeInstance/mongod-cfgsvr-1.tf [hunk 3]
```diff
@@ -8,8 +8,10 @@ resource "google_compute_instance" "mongod_cfgsvr_1" {
     "env"  = "prod"
     "type" = "mongo-config-server"
   }
+  
+  allow_stopping_for_update = true
 
-  machine_type = "e2-highmem-2"
+  machine_type = "e2-standard-2"
```
Comment (123MwanjeMike) [inline infra/terraform/gcp/modules/ComputeInstance/mongod-cfgsvr-1.tf:14]: [downgrade config servers to 'e2-standard-2'](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/4ba14dcab7cf21af03ad7b91995cb13bf0659bcb)

File: infra/terraform/gcp/modules/ComputeDisk/airqo-dev-haproxy.tf
Comment (123MwanjeMike) [inline infra/terraform/gcp/modules/ComputeDisk/airqo-dev-haproxy.tf:1]: [Remove dev sensitive stateful resources](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/35c1f7884d6c0667519815308a665d08da6e1ffb)

File: k8s/nginx/development/global-config.yaml
Comment (123MwanjeMike) [inline k8s/nginx/development/global-config.yaml:1]: [dev nginx virtual servers and global config](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/fc087057933b3332a12df99c671b2c8ef9d7353f)

File: k8s/nginx/values.yaml
Comment (123MwanjeMike) [inline k8s/nginx/values.yaml:1]: [custom values for nginx installation](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/214eab6e21149239316acf09d7b8678eabfcfba2)

**Review by Baalmart — APPROVED** [review]
thanks @123MwanjeMike , this looks good to me.

Just highlighting that we need to ensure smooth operation of all infra services after this PR is fully approved and merged.

Thanks again!

File: k8s/nginx/README.md
Comment (123MwanjeMike) [inline k8s/nginx/README.md:1]: [add nginxinc helm chart installation files](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/80d99974962d59dfc9457a8a4bc106d28d566b50)

File: k8s/nginx/production/docs-vs.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: docs-virtual-server
   namespace: production
 spec:
-  host: docs.airqo.net
+  host: wiki.airqo.net
```
Comment (123MwanjeMike) [inline k8s/nginx/production/docs-vs.yaml:7]: [switch docs subdomains to wiki](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/bd663a8b432a452925f4de7151f745b2146c223a)

File: k8s/nginx/tls-secret.yaml
Comment (123MwanjeMike) [inline k8s/nginx/tls-secret.yaml:1]: Add [tls-secret template](https://github.com/airqo-platform/AirQo-api/pull/1642/commits/5c842e8f50fbf1998fb3dcc0387ad4cab08d90af) for nginx

**Review by 123MwanjeMike — COMMENTED** [review]
Guiding comments

**Comment by 123MwanjeMike** [comment]
> thanks @123MwanjeMike , this looks good to me.
> 
> Just highlighting that we need to ensure smooth operation of all infra services after this PR is fully approved and merged.
> 
> Thanks again!

Thanks @Baalmart

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Pull request merged**